### PR TITLE
Make all db transactions async

### PR DESCRIPTION
**Pull request by freesig**
### Summary
There were some db transactions that were not waiting for permits or running on threads so I fixed that.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain/src/conductor/handle.rs [hunk 1]
```diff
@@ -633,10 +635,15 @@ impl<DS: DnaStore + 'static> ConductorHandleT for ConductorHandleImpl<DS> {
...
-                let res = env
-                    .conn()?
-                    .p2p_gossip_query_agents(since_ms, until_ms, (*arc_set).clone())
-                    .map_err(holochain_p2p::HolochainP2pError::other);
+                let permit = env.conn_permit().await;
+                let res = tokio::task::spawn_blocking(move || {
+                    let mut conn = env.from_permit(permit)?;
+                    conn.p2p_gossip_query_agents(since_ms, until_ms, (*arc_set).clone())
+                })
+                .await;
```
Comment (neonphog) [inline crates/holochain/src/conductor/handle.rs:643]: Is this not something that can be done inside `p2p_gossip_query_agents` so we don't forget to do it in other places?

**Review by neonphog — APPROVED** [review]
:+1:

File: crates/holochain/src/conductor/handle.rs (hunk 1 above)
Comment (maackle) [reply crates/holochain/src/conductor/handle.rs:643]: maybe different types for a conn that waits for a permit vs one that doesn't? and then only define `p2p_gossip_query_agents` on the one that does?

File: crates/holochain_sqlite/src/db/p2p_agent_store.rs [hunk 2]
```diff
@@ -54,7 +54,7 @@ pub trait AsP2pStateTxExt {
     fn p2p_query_near_basis(&self, basis: u32, limit: u32) -> DatabaseResult<Vec<AgentInfoSigned>>;
 }
 
-impl AsP2pAgentStoreConExt for crate::db::PConn {
+impl AsP2pAgentStoreConExt for crate::db::PConnGuard {
```
Comment (maackle) [inline crates/holochain_sqlite/src/db/p2p_agent_store.rs:57]: In regards to my comment about different types...perhaps here is where you'd split these methods into different traits depending on what kind of connection you want to use with them.

**Review by maackle — APPROVED** [review]

File: crates/holochain/src/conductor/handle.rs (hunk 1 above)
Comment (freesig) [reply crates/holochain/src/conductor/handle.rs:643]: That's what is happening. It's only implemented for the `PConnGuard`

File: crates/holochain_sqlite/src/db/p2p_agent_store.rs (hunk 2 above)
Comment (freesig) [reply crates/holochain_sqlite/src/db/p2p_agent_store.rs:57]: Doesn't this already force them to use the `PConnGuard` which can only be created via a permit?

**Pull request merged**